packages/api/lambdas/process-s3-dead-letter-archive.js

```diff
@@ -19,12 +18,22 @@ const { writeRecords } = require('./sf-event-sqs-to-db-records');
  * @returns {string}
  */
 const generateNewArchiveKeyForFailedMessage = (failedKey) => {
-  // Split key with format `${process.env.stackName}/dead-letter-archive/sqs/${messageId}`,
-  const splitKey = failedKey.split('/sqs/');
-  const pathPrefix = splitKey[0];
-  const messageId = splitKey[1];
-  const date = moment.utc().format('YYYY-MM-DD');
-  return `${pathPrefix}/failed-sqs/${date}/${messageId}`;
+  if (failedKey.includes('sqs/')) {
+    /* this is the expected path
+    * this *should* be called with format
+    * `${process.env.stackName}/dead-letter-archive/sqs/${messageId}`
+    * this path will also handle a call pointed to
+    * `${process.env.stackName}/dead-letter-archive/failed-sqs/${messageId}`
+    * */
+    return failedKey.replace('sqs/', 'failed-sqs/');
+  }
+  if (failedKey.includes('dead-letter-archive/')) {
+    // in case a weird path is passed, hopefully it contains dead-letter-archive/
+    return failedKey.replace('failed-dead-letter-archive/');
```

Comment:
What is `failed-dead-letter-archive`?  We have never used this path.

Reply:
I think no matter where the input messages come from, we always put failed messages to `${pathPrefix}/failed-sqs/${date}/${messageId}`;
date should be original eventDate, not the time the message is reprocessed

Reply:
#1: it is using the original eventDate, not the time the message is reprocessed
#2: I had envisioned this as a failure case, that it will only go to these other paths if it *can't* parse the original path such that there is a clear ..../failed-sqs/.... to send it to.
if I'm going to guarantee that it will go into failed-sqs/ but *cannot* parse the /sqs/ folder from the path, can I guarantee that I'll be able to parse date and messageId?

Reply:
We can't put to `${process.env.stackName}/dead-letter-archive/failed-sqs/${date}/${messageId}` regardless the original path

Reply:
Operator can pass in different path, and there is no guarantee the path will have /sqs/.  So to make things simple, we can put the failed messages to predefined location, so the messages won't end up everywhere.